can_nrf failing to build on nrf9280pdk

<!--
**Notes**
Github Discussions (https://github.com/zephyrproject-rtos/zephyr/discussions)
are available to first verify that the issue is a genuine Zephyr bug and not a
consequence of Zephyr services misuse.

This issue list is only for bugs in the main Zephyr code base
(https://github.com/zephyrproject-rtos/zephyr/). If the bug is for a project
fork (such as NCS) specific feature, please open an issue in the fork project
instead.
-->

**Describe the bug**
<!--
A clear and concise description of what the bug is.

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using?
 - What have you tried to diagnose or workaround this issue?
 - Is this a regression? If yes, have you been able to "git bisect" it to a
   specific commit?
 - ...
-->

It would appear as if CAN is not working on nrf9280pdk, with can_nrf.c driver failing to compile (issue seems related to clock control)

**To Reproduce**
<!--
Steps to reproduce the behavior:
1. mkdir build; cd build
2. cmake -DBOARD=board\_xyz
3. make
4. See error
-->

```
west twister -p nrf9280pdk/nrf9280/cpuapp -s canbus.isotp.conformance -i
west twister -p nrf9280pdk/nrf9280/cpuapp -s canbus.isotp.conformance.fd.txdl_32 -i
west twister -p nrf9280pdk/nrf9280/cpuapp -s canbus.isotp.conformance.fd.txdl_64 -i
west twister -p nrf9280pdk/nrf9280/cpuapp -s canbus.isotp.conformance.fd.unused -i
west twister -p nrf9280pdk/nrf9280/cpuapp -s canbus.isotp.implementation -i
west twister -p nrf9280pdk/nrf9280/cpuapp -s drivers.can.api -i
west twister -p nrf9280pdk/nrf9280/cpuapp -s drivers.can.api.rtr -i
west twister -p nrf9280pdk/nrf9280/cpuapp -s drivers.can.host -i
west twister -p nrf9280pdk/nrf9280/cpuapp -s drivers.can.shell -i
west twister -p nrf9280pdk/nrf9280/cpuapp -s drivers.can.timing -i
west twister -p nrf9280pdk/nrf9280/cpuapp -s sample.drivers.can.babbling -i
west twister -p nrf9280pdk/nrf9280/cpuapp -s sample.drivers.can.babbling -i
west twister -p nrf9280pdk/nrf9280/cpuapp -s sample.drivers.can.counter -i
west twister -p nrf9280pdk/nrf9280/cpuapp -s sample.drivers.can.counter -i
west twister -p nrf9280pdk/nrf9280/cpuapp -s sample.modules.canopennode.no_storage -i
west twister -p nrf9280pdk/nrf9280/cpuapp -s sample.modules.canopennode.no_storage -i
west twister -p nrf9280pdk/nrf9280/cpuapp -s sample.net.sockets.can.one_socket -i
west twister -p nrf9280pdk/nrf9280/cpuapp -s sample.net.sockets.can.two_sockets -i
west twister -p nrf9280pdk/nrf9280/cpuapp -s sample.subsys.canbus.isotp -i
```

**Expected behavior**
<!--
A clear and concise description of what you expected to happen.
-->

CAN working / tests passing

**Impact**
<!--
What impact does this issue have on your progress (e.g., annoyance, showstopper)
-->

Test failures in weekly CI / Users probably unable to use CAN on this board (/ SoC?)

**Logs and console output**
<!--
If applicable, add console logs or other types of debug information
e.g Wireshark capture or Logic analyzer capture (upload in zip archive).
copy-and-paste text and put a code fence (\`\`\`) before and after, to help
explain the issue. (if unable to obtain text log, add a screenshot)
-->

```log
/Users/kartben/zephyrproject/zephyr/drivers/can/can_nrf.c: In function 'configure_hsfll':
/Users/kartben/zephyrproject/zephyr/drivers/can/can_nrf.c:140:16: error: variable 'spec' has initializer but incomplete type
  140 |         struct nrf_clock_spec spec = { 0 };
      |                ^~~~~~~~~~~~~~
/Users/kartben/zephyrproject/zephyr/drivers/can/can_nrf.c:140:40: warning: excess elements in struct initializer
  140 |         struct nrf_clock_spec spec = { 0 };
      |                                        ^
/Users/kartben/zephyrproject/zephyr/drivers/can/can_nrf.c:140:40: note: (near initialization for 'spec')
/Users/kartben/zephyrproject/zephyr/drivers/can/can_nrf.c:140:31: error: storage size of 'spec' isn't known
  140 |         struct nrf_clock_spec spec = { 0 };
      |                               ^~~~
/Users/kartben/zephyrproject/zephyr/drivers/can/can_nrf.c:152:16: warning: implicit declaration of function 'nrf_clock_control_request_sync' [-Wimplicit-function-declaration]
  152 |         return nrf_clock_control_request_sync(config->hsfll, &spec, K_FOREVER);
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/kartben/zephyrproject/zephyr/drivers/can/can_nrf.c:140:31: warning: unused variable 'spec' [-Wunused-variable]
  140 |         struct nrf_clock_spec spec = { 0 };
      |                               ^~~~
/Users/kartben/zephyrproject/zephyr/drivers/can/can_nrf.c:153:1: warning: control reaches end of non-void function [-Wreturn-type]
  153 | }
      | ^
```

**Environment (please complete the following information):**

 - OS: (e.g. Linux, MacOS, Windows): macOS
 - Toolchain (e.g Zephyr SDK, ...): 0.17.0
 - Commit SHA or Version used: 94cd7865a99d4c19a7e33dc4cd48b9bb9d70ac42

**Additional context**
<!--
Add any other context that could be relevant to your issue, such as pin setting,
target configuration, ...
-->

Bissected to d78952b7ba930e20910c2b6fdac49cc8dc4f709e (PR #82334 by @gmarull)

--- a/boards/nordic/nrf9280pdk/nrf9280pdk_nrf9280-pinctrl.dtsi
+++ b/boards/nordic/nrf9280pdk/nrf9280pdk_nrf9280-pinctrl.dtsi
@@ -60,13 +60,6 @@
 		};
 	};
 
-	/omit-if-no-ref/ can120_default: can120_default {
-		group1 {
-			psels = <NRF_PSEL(CAN_RX, 9, 4)>,
-				<NRF_PSEL(CAN_TX, 9, 5)>;
-		};
-	};
-
 	/omit-if-no-ref/ pwm130_default: pwm130_default {
 		group1 {
 			psels = <NRF_PSEL(PWM_OUT0, 9, 2)>;

--- a/boards/nordic/nrf9280pdk/nrf9280pdk_nrf9280_cpuapp.dts
+++ b/boards/nordic/nrf9280pdk/nrf9280pdk_nrf9280_cpuapp.dts
@@ -27,7 +27,6 @@
 		zephyr,ieee802154 = &cpuapp_ieee802154;
 		zephyr,bt-hci = &bt_hci_ipc0;
 		nordic,802154-spinel-ipc = &ipc0;
-		zephyr,canbus = &can120;
 	};
 
 	aliases {
@@ -273,16 +272,6 @@ ipc0: &cpuapp_cpurad_ipc {
 
 zephyr_udc0: &usbhs {
 	status = "okay";
-};
-
-&canpll {
-	status = "okay";
-};
-
-&can120 {
-	status = "okay";
-	pinctrl-0 = <&can120_default>;
-	pinctrl-names = "default";
 };
 
 &pwm130 {

--- a/boards/nordic/nrf9280pdk/nrf9280pdk_nrf9280_cpuapp.yaml
+++ b/boards/nordic/nrf9280pdk/nrf9280pdk_nrf9280_cpuapp.yaml
@@ -13,7 +13,6 @@ ram: 512
 flash: 1024
 supported:
   - adc
-  - can
   - counter
   - gpio
   - i2c

--- a/dts/common/nordic/nrf9280.dtsi
+++ b/dts/common/nordic/nrf9280.dtsi
@@ -445,28 +445,6 @@
 				global-domain-id = <12>;
 			};
 
-			can120: can@8d8000 {
-				compatible = "nordic,nrf-can";
-				reg = <0x8d8000 0x400>, <0x2fbef800 0x800>, <0x2fbe8000 0x7800>;
-				reg-names = "wrapper", "m_can", "message_ram";
-				interrupts = <216 NRF_DEFAULT_IRQ_PRIORITY>;
-				clocks = <&canpll>, <&hsfll120>;
-				clock-names = "auxpll", "hsfll";
-				bosch,mram-cfg = <0x0 28 8 3 3 0 1 1>;
-				status = "disabled";
-			};
-
-			can121: can@8db000 {
-				compatible = "nordic,nrf-can";
-				reg = <0x8db000 0x400>, <0x2fbf7800 0x800>, <0x2fbf0000 0x7800>;
-				reg-names = "wrapper", "m_can", "message_ram";
-				interrupts = <219 NRF_DEFAULT_IRQ_PRIORITY>;
-				clocks = <&canpll>, <&hsfll120>;
-				clock-names = "auxpll", "hsfll";
-				bosch,mram-cfg = <0x0 28 8 3 3 0 1 1>;
-				status = "disabled";
-			};
-
 			dppic120: dppic@8e1000 {
 				compatible = "nordic,nrf-dppic-global";
 				reg = <0x8e1000 0x1000>;

--- a/tests/kernel/gen_isr_table/testcase.yaml
+++ b/tests/kernel/gen_isr_table/testcase.yaml
@@ -72,6 +72,8 @@ tests:
     platform_exclude:
       - m2gl025_miv
       - adp_xc7k/ae350
+      - nrf9280pdk/nrf9280/cpuppr
+      - nrf9280pdk/nrf9280/cpuppr/xip
     filter: CONFIG_RISCV_PRIVILEGED
     extra_configs:
       - CONFIG_GEN_IRQ_VECTOR_TABLE=y

